src/transformers/core/createFilter.ts

```diff
@@ -1,13 +1,23 @@
-import { Filter } from '../../models/searchservice/request/Filter';
+import { Filter, NearFilterValue } from '../../models/searchservice/request/Filter';
 import { Matcher } from '../../models/searchservice/common/Matcher';
+import { NumberRangeValue } from '../../models/searchservice/common/NumberRangeValue';
 
 export function createFilter(filter: any): Filter {
   const fieldId = Object.keys(filter)[0];
   const matcher = Object.keys(filter[fieldId])[0];
-
+  const rawValue = filter[fieldId][matcher];
+  let value: string | number | boolean | NearFilterValue | NumberRangeValue;
+  if (matcher === Matcher.Between) {
+    value = {
+      start: { matcher: Object.keys(rawValue)[0], value: Object.values(rawValue)[0] },
```

Comment:
Is the first key guaranteed to be the start matcher? Even so, it's probably a good idea to some checking so that we aren't relying on key ordering